buffer.c

```diff
@@ -105,6 +105,8 @@ static int pv_buffer_init_cache(int items, int size, struct dl_list *head)
 			free(item);
 		}
 	}
+
+	dl_list_init(head);
```

Comment:
It amazes me that this wasn't initialized all this time... The variable head is still used previously in this function though. Don't get me wrong, I think this is the place for this dl_list_init, but the previous code should be moved to pv_buffer_free or similar and I don't understand what was this doing here.

Seeing buffer in a general way, as it is used by metadata, ph_logger and logserver, maybe it shouldn't be initialized from log.c either and moved it to a more general location. Besides this, the new pv_buffer_free would have to be called somewhere at the end of pv_shutdown so we can fix this mem leak.

If you think it's a lot for this PR, we can leave it for a further one.

Reply:
Yeah, probably we need a new PR to fix that maybe pv_buffer_init() and pv_buffer_free() I think that both must be called by any file or library that needs a buffer and internally the buffer could work with reference counting, so if it's already initialized pv_buffer_init() does nothing except add 1 to the ref counter. If pv_buffer_free() is called decrease the counter, if the counter is equals to 0 run the routine to free all the resources.